lib/semantics/tools.cc

```diff
@@ -328,6 +331,29 @@ const Symbol *FindFunctionResult(const Symbol &symbol) {
   return nullptr;
 }
 
+static const Symbol *GetAssociatedVariable(const AssocEntityDetails &details) {
+  if (const MaybeExpr & expr{details.expr()}) {
+    if (evaluate::IsVariable(*expr)) {
+      if (const Symbol * varSymbol{evaluate::GetLastSymbol(*expr)}) {
+        return GetAssociationRoot(*varSymbol);
+      }
+    }
+  }
+  return nullptr;
+}
+
+// Return the Symbol of the variable of a construct association, if it exists
+// Return nullptr if the name is associated with an expression
```

Comment:
I think this comment needs to be updated.

Reply:
Could you be more specific about the change you want?  Is the comment not understandable, or is it incorrect in some respect?

Reply:
`GetAssociationRoot()` [used to have](https://github.com/flang-compiler/f18/pull/596/commits/c9d2507b74da881dda2eb11805a0394a415db2e4#diff-1f02100eb6511d1af252e0a76b91ecafR353) a `return nullptr;` in your original commit of this PR. With your follow up commits, the structure of this function changed and so the comment about `return nullptr` doesn't seem to fit.

Reply:
Right.  The code that used to return a ```nullptr``` was pulled out and put into the static function ```GetAssociatedVariable()```.  But the effect of this is that ```GetAssociationRoot()```, which is the only externally visible function, still either returns a ```nullptr``` in the case where its parameter is construct associated with an expression.

Reply:
That makes sense, thanks for the explanation.